pkg/resources/validation/podsecurityadmissionconfigurationtemplate/podsecurityact.go

```diff
@@ -0,0 +1,295 @@
+package podsecurityadmissionconfigurationtemplate
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	mgmtv3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	provv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/webhook/pkg/admission"
+	"github.com/rancher/webhook/pkg/clients"
+	v3 "github.com/rancher/webhook/pkg/generated/controllers/management.cattle.io/v3"
+	v1 "github.com/rancher/webhook/pkg/generated/controllers/provisioning.cattle.io/v1"
+	objectsv3 "github.com/rancher/webhook/pkg/generated/objects/management.cattle.io/v3"
+	admissionv1 "k8s.io/api/admission/v1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	machinery "k8s.io/apimachinery/pkg/api/validation"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/pod-security-admission/api"
+	"k8s.io/utils/trace"
+)
+
+var GVR = schema.GroupVersionResource{
+	Group:    "management.cattle.io",
+	Version:  "v3",
+	Resource: "podsecurityadmissionconfigurationtemplates",
+}
+
+type Validator struct {
+	ManagementClusterCache   v3.ClusterCache
+	provisioningClusterCache v1.ClusterCache
+}
+
+const ByPodSecurityAdmissionConfigurationName = "podSecurityAdmissionConfigurationName"
+
+func NewPodSecurityAdmissionConfigurationTemplateValidator(clients *clients.Clients) *Validator {
+	return &Validator{
+		ManagementClusterCache:   clients.Management.Cluster().Cache(),
+		provisioningClusterCache: clients.Provisioning.Cluster().Cache(),
+	}
+}
+
+func (v Validator) ValidatingWebhook(clientConfig admissionregistrationv1.WebhookClientConfig) *admissionregistrationv1.ValidatingWebhook {
```

Comment:
```suggestion
func (v *Validator) ValidatingWebhook(clientConfig admissionregistrationv1.WebhookClientConfig) *admissionregistrationv1.ValidatingWebhook {
```
This change needs to be expanded to all other methods associated to  the struct